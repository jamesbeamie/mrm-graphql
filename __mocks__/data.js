const data = {
  refetch: jest.fn(),
  office: {
    id: '11',
    name: 'EPIC Tower',
    timezone: 'GMT +1',
    location: {
      timeZone: 'EAST_AFRICA',
    },
  },
  officeII: {
    id: '11',
    name: 'EPIC Tower',
    timezone: 'GMT +1',
    location: {
      timeZone: 'WEST_AFRICA',
    },
  },
  resource: {
    name: 'Marker',
    rooms: 4,
  },
  routes: {
    home: '/',
  },
};
module.exports = data;
