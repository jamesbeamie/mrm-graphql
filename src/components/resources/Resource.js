import React from 'react';
import PropTypes from 'prop-types';
import WrappedDeleteResource from './DeleteResource';
import WrappedEditResource from './EditResource';

/**
 * Component that shows resources in a table
 *
 * @param {Object} resourcesObject
 *
 * @returns {JSX}
 */
const Resource = ({ resource, refetch, currentPage }) => (
  <div className="table__row">
    <span>{resource.name}</span>
    <span>
      <WrappedEditResource resource={resource} refetch={refetch} currentPage={currentPage} /> &nbsp;
      <WrappedDeleteResource toDelete={resource} refetch={refetch} currentPage={currentPage} />
    </span>
  </div>
);

Resource.propTypes = {
  resource: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }).isRequired,
  refetch: PropTypes.func,
  currentPage: PropTypes.number,
};

Resource.defaultProps = {
  refetch: null,
  currentPage: 1,
};

export default Resource;
