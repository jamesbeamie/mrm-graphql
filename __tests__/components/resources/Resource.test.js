import React from 'react';
import { shallow } from 'enzyme';

import Resource from '../../../src/components/resources/Resource';
import WrappedDeleteResource from '../../../src/components/resources/DeleteResource';
import data from '../../../__mocks__/data';


describe('Tests for Resource Component', () => {
  const shallowWrapper = shallow(<Resource resource={data.resource} doDelete={() => {}} />);

  it('renders correctly from memory', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
  it('should render the `DeleteResource`', () => {
    expect(shallowWrapper.find(WrappedDeleteResource)).toHaveLength(1);
  });
});
