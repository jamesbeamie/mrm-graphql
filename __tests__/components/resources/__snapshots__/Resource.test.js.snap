// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Resource Component renders correctly from memory 1`] = `
<div
  className="table__row"
>
  <span>
    Marker
  </span>
  <span>
    <Apollo(Apollo(EditResource))
      currentPage={1}
      refetch={null}
      resource={
        Object {
          "name": "Marker",
          "rooms": 4,
        }
      }
    />
     Â 
    <Apollo(Apollo(DeleteResource))
      currentPage={1}
      refetch={null}
      toDelete={
        Object {
          "name": "Marker",
          "rooms": 4,
        }
      }
    />
  </span>
</div>
`;
